Lib9c/Action/UnlockEquipmentRecipe.cs

```diff
@@ -83,20 +90,34 @@ public override IAccountStateDelta Execute(IActionContext context)
                     throw new AlreadyRecipeUnlockedException($"recipe: {recipeId} already unlocked.");
                 }
 
-                var prevId = recipeId - 1;
-                if (!unlockedIds.Contains(prevId))
+                EquipmentItemRecipeSheet.Row recipeRow = equipmentRecipeSheet[recipeId];
+
+                if (!worldInformation.IsStageCleared(recipeRow.UnlockStage))
                 {
-                    // Can't skip previous recipe unlock.
-                    throw new InvalidRecipeIdException($"unlock {prevId} first.");
+                    throw new NotEnoughClearedStageLevelException($"clear {recipeRow.UnlockStage} first.");
                 }
 
-                unlockedIds.Add(recipeId);
+                EquipmentItemSheet.Row equipmentRow = equipmentItemSheet[recipeRow.ResultEquipmentId];
 
-                EquipmentItemRecipeSheet.Row row = equipmentRecipeSheet[recipeId];
-                if (!worldInformation.IsStageCleared(row.UnlockStage))
+                // Ignore grade 0 for default equipment.
+                int firstId = equipmentItemSheet
+                    .OrderedList
+                    .First(r => r.ItemSubType == equipmentRow.ItemSubType && r.Grade > 0)
+                    .Id;
+                // Check recipe is first row by ItemSubType.
+                if (equipmentRow.Id != firstId)
                 {
-                    throw new NotEnoughClearedStageLevelException($"clear {row.UnlockStage} first.");
+                    var prevId = recipeId - 1;
+                    if (!unlockedIds.Contains(prevId))
+                    {
+                        // Can't skip previous recipe unlock.
+                        throw new InvalidRecipeIdException($"unlock {prevId} first.");
+                    }
                 }
```

Comment:
I think previous recipe is checked via `Row.UnlockStage`, not with `Row.Id`.

Reply:
It is correct to check recipeId because it contains from list of unlocked recipe Ids. we check the stageId at 95line.